### DISCUSSION TITLE
Line graph is weird - what am I doing wrong?

### DISCUSSION THREAD

**Discussion by aleksrutins** [f5/unovis#176] [Q&A] [answered]
Recently, my line charts changed from being normal to looking like this:
![image](https://user-images.githubusercontent.com/24496030/229285373-41d23170-61d9-4bad-aede-5f5b8de49927.png)
(don't mind the scale of the x-axis; that's something I need to work on)

The component I'm rendering the graph with is [VisitsChart.svelte](https://github.com/aleksrutins/litelytics/blob/0258e6f36b4b71d169fc2f46bc158f9c37f9ad09/frontend/src/components/data-display/VisitsChart.svelte):
```svelte
<script lang="ts">
    import { Visit } from "../../api/sites";
    import { VisXYContainer, VisLine, VisAxis } from "@unovis/svelte"


    export let visits: Visit[]

    function visitsToData(visits: Visit[]) {
        let result = new Map<number, number>()
        for(const visit of visits) {
            const day = Math.round(new Date(visit.timestamp).getTime() / (1000 * 60 * 60 * 24 * 7))
            if(result.has(day)) result.set(day, result.get(day)! + 1)
            else result.set(day, 1)
        }
        return Array.from(result.entries());
    }

    $: data = visitsToData(visits)

    const tickFormat = (tick: number) => new Date(tick * 1000 * 60 * 60 * 24 * 7).toDateString()
</script>

<VisXYContainer {data}>
    <VisLine curveType="step" {data} x={d => d[0]} y={d => d[1]} />
    <VisAxis type="x" {tickFormat}/>
    <VisAxis type="y"/>
</VisXYContainer>
```

What am I doing that's making it this weird?

(and no, this is not an April Fools joke)

**Comment by rokotyan** [comment, marked as answer]
@aleksrutins The first thing I would do is check whether the data is sorted chronologically. Unovis doesn't sort data by itself.

**Reply by aleksrutins** [reply]
That was the problem, thanks!
